packages/blade/src/components/ActionList/ActionListFooter.tsx

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/prefer-ts-expect-error */
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+
+import React from 'react';
+import styled from 'styled-components';
+import { componentIds } from './componentIds';
+import Box from '~components/Box';
+import type { IconComponent } from '~components/Icons';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import { isReactNative, makeSize } from '~utils';
+import { Text } from '~components/Typography';
+
+type ActionListFooterProps = {
+  title?: string;
+  leading?: React.ReactNode;
+  trailing?: React.ReactNode;
+};
+
+const StyledActionListFooter = styled(Box)((props) => {
+  return {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    padding: `${makeSize(props.theme.spacing[3])} ${makeSize(props.theme.spacing[5])}`,
+    backgroundColor: props.theme.colors.brand.gray.a50.lowContrast,
+    margin: `${makeSize(props.theme.spacing[3])} -${makeSize(props.theme.spacing[3])} -${makeSize(
+      props.theme.spacing[3],
+    )}`,
+  };
+});
+
+const ActionListFooter = (props: ActionListFooterProps): JSX.Element => {
+  const footerRef = React.useRef<HTMLDivElement | null>(null);
+  const {
+    setShouldIgnoreBlur,
+    setHasFooterAction,
+    onSelectKeydown,
+    activeIndex,
+    setIsOpen,
+  } = useDropdown();
+
+  React.useEffect(() => {
+    if (!isReactNative() && footerRef.current?.querySelector('button, a')) {
+      setHasFooterAction(true);
+    }
+  }, [setHasFooterAction, props.trailing]);
+
+  const isOnlyActionButton = !props.title && !props.leading && props.trailing;
+
+  return (
+    <StyledActionListFooter
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ref={footerRef as any}
+      // @ts-ignore: Ignoring because the TS fails for React Native and works for web
+      onMouseDown={() => {
```

Comment:
If you mean that flicker of animation, it is happening because the blur of input gets triggered and we focus on it again. 

This blur functionality in particular doesn't really change the `BaseInput`'s blur behaviour. It only changes the blur functionality that dropdown controls.

To fix the flicker, BaseInput will need some refactor to support ignoring the animation for the input blur. Lemme look into once again.